### PR TITLE
Disk adapters 

### PR DESCRIPTION
**Pull request by dan-loomb**

### CODE REVIEW

File: src/Adapters/IConfigAdapter.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Statsig\Adapters;
+
+interface IConfigAdapter
+{
+    /**
+     * @return array<string, array>
+     */
+    function getConfigSpecs(): array;
```
Comment (tore-statsig) [inline src/Adapters/IConfigAdapter.php]: You're going to need a set and a get

**Review by tore-statsig — COMMENTED** [review]
Awesome start!

High level, I think the pattern here is:

Check for config specs from the adapter.  If you get them, great!  Evaluate.  If not, call Statsig, and cache it in the provided adapter.  Then evaluate.

It looks like this still inherits from the file storage solution and assumes the storage is populated from some other job (like the sync file).  In an ideal world, you dont even need the sync file.

**Review by tore-statsig — COMMENTED** [review]
What about send.php and sync.php

File: src/Adapters/IConfigAdapter.php (hunk 1 above)
Comment (dan-loomb) [reply src/Adapters/IConfigAdapter.php]: Added.

**Comment by dan-loomb** [comment]
Updated to be able to provide an Adapter for the sync/send jobs. Added fallback to downloadConfigSpecs if the spec file has not been loaded.

**Comment by dan-loomb** [comment]
Merged on private repo

**Pull request closed without merging**